jbpm-flow/src/main/java/org/jbpm/process/core/context/variable/Variable.java

```diff
@@ -80,6 +90,9 @@ public void setValue(final Object value) {
 
     public void setMetaData(String name, Object value) {
         this.metaData.put(name, value);
+        if (VARIABLE_TAGS.equals(name) && value != null) {
```

Comment:
It seems like "setMetaData" is never called for "customTags" name, so tags is never cached, if this was the purpose of these lines.

Reply:
look at the bpmn2. you will understand the customTags

Reply:
My point is that in the PropertyHandler class (correct me if I'm wrong), it's missing some code to read the metaData, and in case of "customTags" is present, then invoke setMetaData with that key.

Reply:
this is set via org.jbpm.bpmn2.xml.MetaDataHandler.MetaDataWrapper

Reply:
But this condition is never matched:
VARIABLE_TAGS.equals(name)
Is this to cache tags?

Reply:
this is an extra way of setting tags that can be done programatically like fluent api so it should be handled too.

Reply:
ok, thanks for the clarification